GitUI/AutoCompletion/CommitMessageMetadataProvider.cs

```diff
@@ -0,0 +1,19 @@
+﻿namespace GitUI.AutoCompletion;
+
+internal class CommitMessageMetadataProvider : IAutoCompleteProvider
+{
+    private static readonly string[] keywords = [
+        "Co-authored-by: ",
+        "Signed-off-by: ",
+        "BREAKING CHANGE: ",
+        "Reviewed-by: ",
+        "Tested-by: ",
+        ];
+
+    private static readonly AutoCompleteWord[] _autoCompleteWords = keywords.Select(k => new AutoCompleteWord(k)).ToArray();
+
+    public Task<IEnumerable<AutoCompleteWord>> GetAutoCompleteWordsAsync(CancellationToken cancellationToken)
```

Comment:
Have you given any thought to using [`ImmutableArray`](https://learn.microsoft.com/dotnet/api/system.collections.immutable.immutablearray)? We can bump the version to 8.0.

Reply:
Version is bumped to C#12 already in 1cfb0e4441e260ff252f258b08cb403df76801e5